internal/config/config.go

```diff
@@ -179,3 +182,23 @@ type saveError struct {
 func (e saveError) Error() string {
 	return fmt.Sprintf("unable to save the configuration: %v", e.err)
 }
+
+// IsProjectVersionSupported returns true if the project is already configured and it is v2
+func IsProjectVersionSupported() (bool, error) {
```

Comment:
This function is a little weird as-is. The `bool` return value is never used, and returning no error for a non-existent config doesn't make sense. The following is more succinct, and should be used only once in [`pkg/cli/cli.initialize()`](https://github.com/kubernetes-sigs/kubebuilder/blob/master/pkg/cli/cli.go#L195):
```suggestion
func (c Config) CheckProjectVersionSupported() error {
	if c.IsV2() || c.IsV3() {
		return nil
	}
	return fmt.Errorf(NoticeColor, "Project version "+c.Version+" is not supported. "+
		"See how to upgrade your project to the latest version: https://book.kubebuilder.io/migration/guide.html")
}
```

Reply:
> This function is a little weird as-is. The bool return value is never used, and returning no error for a non-existent config doesn't make sense.

Agreed. 

> The following is more succinct, and should be used only once in pkg/cli/cli.initialize():

Follow up your suggestion but kept the Read because it shows not part of the model. For your fully suggestion work we would need to move it for there.

Reply:
Why wouldn't this work in `internal/config`?

Reply:
See that we would need to do the Read outside as well. Also, after check it, I realize that has parts that are using the helpers in the internal and others not. So, IMO it requires a specific cleanup. Could we agree in do it in a follow-up?

Reply:
reverted the helper as we chat about.